containers/src/Kokkos_DynamicView.hpp

```diff
@@ -821,12 +824,13 @@ inline void deep_copy(const Kokkos::Experimental::DynamicView<T, DP...>& dst,
   else if (SrcExecCanAccessDst)
     Kokkos::Impl::ViewRemap<dst_type, src_type, src_execution_space>(dst, src);
   else
-    src.impl_get_chunks().deep_copy_to(dst.impl_get_chunks());
+    src.impl_get_chunks().deep_copy_to(dst_execution_space{},
+                                       dst.impl_get_chunks());
   Kokkos::fence("Kokkos::deep_copy(DynamicView)");
 }
 
 template <class ExecutionSpace, class T, class... DP, class... SP>
-inline void deep_copy(const ExecutionSpace&,
+inline void deep_copy(const ExecutionSpace& exec,
```

Comment:
Adding execution space support here is helpful to https://github.com/DARMA-tasking/magistrate/pull/202/ so thanks!

Reply:
Also advances #4793